server/src/main/java/org/candlepin/resource/util/ResolverUtil.java

```diff
@@ -78,35 +80,57 @@ public Owner resolveOwner(Owner owner) {
         return owner;
     }
 
-    public Product resolveProduct(Owner owner, Product product) {
-        String id = null;
-        if (product != null) {
-            id = product.getId();
-        }
-        return resolveProduct(owner, id);
+    public Product resolveProduct(Owner owner, String productId, boolean allowNull) {
+        Product product = new Product();
```

Comment:
Rather than create new Product instances, I would much rather change the resolveProduct such that we break it up into two methods (resolveProductById, resolveProductByUuid) or takes the ID and UUID as string params, then flip the Product- and ProductData-based methods use those.